rust/src/scannerctl/syntax/check.rs

```diff
@@ -6,64 +6,42 @@ use std::path::{Path, PathBuf};
 
 use scannerlib::nasl::syntax::load_non_utf8_path;
 use scannerlib::nasl::syntax::{parse, Statement, SyntaxError};
+use scannerlib::nasl::WithErrorInfo;
+use tracing::{debug, error, info};
 use walkdir::WalkDir;
 
-use crate::{CliError, CliErrorKind};
-
-fn read_errors<P: AsRef<Path>>(path: P) -> Result<Vec<SyntaxError>, CliErrorKind> {
-    let code = load_non_utf8_path(path.as_ref())?;
-    Ok(parse(&code)
-        .filter_map(|r| match r {
-            Ok(_) => None,
-            Err(err) => Some(err),
-        })
-        .collect())
-}
+use crate::{CliError, CliErrorKind, Filename};
 
 fn read<P: AsRef<Path>>(path: P) -> Result<Vec<Result<Statement, SyntaxError>>, CliErrorKind> {
     let code = load_non_utf8_path(path.as_ref())?;
     Ok(parse(&code).collect())
 }
 
-fn print_results(path: &Path, verbose: bool) -> Result<usize, CliError> {
-    let mut errors = 0;
+fn print_results(path: &Path) -> Result<usize, CliError> {
+    let mut num_errors = 0;
 
-    if verbose {
-        println!("# {path:?}");
-        let results = read(path).map_err(|kind| CliError {
-            kind,
-            filename: format!("{path:?}"),
-        })?;
-        for r in results {
-            match r {
-                Ok(stmt) => println!("{stmt:?}"),
-                Err(err) => eprintln!("{err}"),
+    println!("# {path:?}");
+    let results = read(path).map_err(|e| e.with(Filename(path)))?;
+    for r in results {
+        match r {
+            Ok(stmt) => debug!("{stmt:?}"),
+            Err(err) => {
+                num_errors += 1;
+                error!("{err}")
             }
         }
-    } else {
-        let err = read_errors(path).map_err(|kind| CliError {
-            kind,
-            filename: format!("{path:?}"),
-        })?;
-        if !err.is_empty() {
-            eprintln!("# Error in {path:?}");
-        }
-        errors += err.len();
-        err.iter().for_each(|r| eprintln!("{r}"));
     }
-    Ok(errors)
+    Ok(num_errors)
 }
 
-pub fn run(path: &PathBuf, verbose: bool, no_progress: bool) -> Result<(), CliError> {
+pub fn run(path: &PathBuf) -> Result<(), CliError> {
     let mut parsed: usize = 0;
     let mut skipped: usize = 0;
     let mut errors: usize = 0;
-    println!("verifying NASL syntax in {path:?}.");
+    info!("Verifying NASL syntax in {path:?}.");
```

Comment:
I don't like the idea of using a logger here as it should be easy to read stdout output and the extra information of a logger doesn't make sense to me as it is called as a cli and not within a service.